org.eclipse.languageserver/src/org/eclipse/languageserver/LanguageServiceAccessor.java

```diff
@@ -111,14 +114,34 @@ public boolean equals(Object obj) {
 		public final URI fileUri;
 		public final IFile file;
 		public final IDocument document;
-		public final LanguageClientEndpoint languageClient;
+		private final LanguageClientEndpoint languageClient;
 
 		public LSPDocumentInfo(URI fileUri, IFile file, IDocument document, LanguageClientEndpoint languageClient) {
 			this.fileUri = fileUri;
 			this.file = file;
 			this.document = document;
 			this.languageClient = languageClient;
 		}
```

Comment:
As the code looke like now, if the editor input is not an IFileEditorInput or an IURIEditorInput, then the document may be null. Can that happen in practice?